src/agent/input-tester/src/crash_detector.rs

```diff
@@ -313,26 +270,43 @@ impl<'a> DebugEventHandler for CrashDetectorEventHandler<'a> {
         }
     }
 
-    fn on_breakpoint(&mut self, _debugger: &mut Debugger, id: BreakpointId) {
-        if let Some(id_to_block) = &self.id_to_block {
-            if let Some(&(mod_idx, block_idx)) = id_to_block.get(&id) {
-                if let Some(coverage_map) = &mut self.coverage_map {
-                    coverage_map.report_block_hit(mod_idx, block_idx);
-                }
+    fn on_create_process(&mut self, dbg: &mut Debugger, module: &Module) {
+        if let Some(coverage) = &mut self.coverage {
+            if let Err(err) = coverage.on_create_process(dbg, module) {
+                error!("error recording coverage on create process: {:?}", err);
+                dbg.quit_debugging();
+            }
+        }
+    }
+
+    fn on_load_dll(&mut self, dbg: &mut Debugger, module: &Module) {
+        if let Some(coverage) = &mut self.coverage {
+            if let Err(err) = coverage.on_load_dll(dbg, module) {
+                error!("error recording coverage on load DLL: {:?}", err);
+                dbg.quit_debugging();
+            }
+        }
+    }
+
+    fn on_breakpoint(&mut self, dbg: &mut Debugger, bp: BreakpointId) {
+        if let Some(coverage) = &mut self.coverage {
+            if let Err(err) = coverage.on_breakpoint(dbg, bp) {
+                error!("error recording coverage on breakpoint: {:?}", err);
+                dbg.quit_debugging();
             }
         }
     }
 }
 
 /// This function runs the application under a debugger to detect any crashes in
 /// the process or any children processes.
-pub fn test_process(
+pub fn test_process<'a>(
     app_path: impl AsRef<OsStr>,
     args: &[impl AsRef<OsStr>],
     env: &HashMap<String, String>,
     max_duration: Duration,
     ignore_first_chance_exceptions: bool,
-    coverage_map: Option<&mut AppCoverageBlocks>,
+    cache: Option<&'a mut ModuleCache>,
```

Comment:
Assumes we retain just this module disassembly cache in-memory across runs, or reload it from disk.